### PR TITLE
Update golang to 1.22.7

### PR DESCRIPTION
**Pull request by cheftako**
Pending dependency changes require golang 1.22.7.
Upgrading to allow those dependencies to go through.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/apiserver-network-proxy
 
-go 1.22.5
+go 1.22.7
```
Comment (jkh52) [inline go.mod:3]: This file lost a "toolchain" directive fairly recently, should we restore it?

**Comment by jkh52** [comment]
/lgtm
/hold

Hold for the toolchain question. Feel free to punt.

File: go.mod (hunk 1 above)
Comment (cheftako) [reply go.mod:3]: I'd like to get it back here and in the matching test-infra repo. However right now I'd like to unblock dependabot.

**Pull request merged**